src/hooks/useRedirectMaintenance.ts

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useRef } from 'react';
+
+import router from '@/router';
+import { PathValues } from '@/types/ui/path';
+
+const TIME_THRESHOLD = 10 * 60 * 1000;
+
+const useRedirectMaintenance = (startHours: number, endHours: number, redirectPath: PathValues) => {
+  const timeoutRef = useRef<number | null>(null);
+
+  useEffect(() => {
+    const checkAndRedirect = () => {
+      const now = new Date();
+      const hours = now.getHours();
+
+      if (hours >= startHours && hours < endHours) {
+        return router.navigate(redirectPath);
+      }
+
+      const nextTime = new Date();
```

Comment:
13line 에서 선언한 시각을 활용해
```suggestion
      const nextTime = new Date(now);
```
로 설정하면 객체 생성이 한 번만 일어나고, 선언 부에 함께 선언할 수 있겠네요!